### DISCUSSION TITLE
Add a frame to the picture

### DISCUSSION THREAD

**Discussion by laguinch** [photobooth-app/photobooth-app#98] [Q&A] [answered]
Hy, thx a lot for the phtobooth app. I want to take picture with a frame. Wich way can I put a frame in the app ? Thx

**Comment by mgineer85** [comment, marked as answer]
Hello! Right now not possible - will implement today later.
You want to add a frame with transparency? The captured image shall blend in through transparent parts of the frame, right?

**Reply by laguinch** [reply]
Yes I want a frame around the captured image exactly

**Reply by laguinch** [reply]
like in the the photobooth project

**Reply by mgineer85** [reply]
I see. I will check later how to implement.

**Reply by laguinch** [reply]
A mother question is ti possible to link 2 different weed's. One for the ring around the camera and an other bigger for a "flash" ?
Thx

**Reply by mgineer85** [reply]
you mean 2 WLEDs?   Can you please provide a bit more description what the second one should do when?

**Reply by laguinch** [reply]
yes 2 wleds one same as the exemple in your install and a second like a flash if it's too dark like a flash photo

**Reply by mgineer85** [reply]
Would a gpio switching on the pi make sense for you also?

**Reply by laguinch** [reply]
I don't know what you mean ?

**Reply by mgineer85** [reply]
Switch a pin on the raspberry pi so you could add a circuit that enables a light source for example.
Not sure whether wled leds are suitable to lighten the scene because it can create flickering

**Reply by laguinch** [reply]
ok,thx

**Comment by mgineer85** [comment]
Updated version released. It's a very basic implementation right now.
Enable the stage in the admin center and provide an image (PNG) as frame. The image shall be stored in the folder "userdata" below the working directory.  Example to set the file: "userdata/frame.png"

**Reply by laguinch** [reply]
did you received  my PayPal donation ?

**Reply by mgineer85** [reply]
I did, thank you very much 😊

**Reply by mgineer85** [reply]
Another version is ready to release soon.
The implementation changed slightly in how the images are arranged. 
If a frame is used the resulting image has same size. The captured image is mounted to transparent area that is detected automatically. That should result in best looking pics 😌

**Reply by laguinch** [reply]
where can I download the original frame to make some personal arrangement. Thx

**Reply by mgineer85** [reply]
You need to provide your own PNG.
You could find it here: https://github.com/mgrl/photobooth-app/tree/main/photobooth/assets/frames

**Reply by laguinch** [reply]
Hy, i have a problem to use my frame. Where must be the file uploaded ? what is the path in the Configure stages how to process images after capture menu ? Thx

**Reply by mgineer85** [reply]
Use the files tab to upload to userdata folder: 
![image](https://github.com/photobooth-app/photobooth-app/assets/3868450/c84141ac-a74f-4139-afd0-e84ceca804cd)

The path to configure stages would be userdata/filename.png (replace by your filename of course).

**Reply by laguinch** [reply]
Hy thx it's ok now :)

**Reply by Striker62** [reply]
Looking for the same. 

found in the Docu

<html><body>

userdata (images, frames, fonts, ...) | /userdata | via adminpanel file explorer
-- | -- | --



</body>
</html>

I've setup my userdata according to the default location of the sample file  structure userdat/images, userdata/frames , userdate/fonts. I but my frame in my frame folder. Now I know why it does not work.

Wold it be a great issue to implent it according to the default folder ? makes it easier to maintain ?

**Reply by mgineer85** [reply]
You can use subfolders, no problem.
Userdata is not deleted in any case, the other folders could be emptied by photobooth from time to time

**Reply by Striker62** [reply]
Sunfower ?

My question was if you can allow and use the userdata Folder like i did 

working directory userdate ---|
                                               |-----Images
                                               |-----frames
                                               |-----fonts

**Reply by mgineer85** [reply]
Sorry autocorrect.
Yes you can use like that 👍
